lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm

```diff
@@ -196,4 +197,26 @@ sub forget_run_last {
     return $self->render(json => {message => "error $!"}, status => 500);
 }
 
+sub latest_test {
+    my $self  = shift;
+    my $alias = $self->param('alias');
+    my $full  = $self->param('full');
+    my $id    = $self->obs_rsync->get_last_test_id($alias);
+    return undef unless $id;
+
+    return $self->render(json => {id => $id}, status => 200) unless $full;
+    my $result = '';
+    my $code   = 500;
+    try {
+        my $job = $self->schema->resultset("Jobs")->single({id => $id});
+        $result = $job->result ? $job->result : 'Not found' if $job;
```

Comment:
The result is not a nullable column and default initialized with `none`. I'm not aware of any place where we set it back to an empty string so I'm wondering what case you want to distinguish using the ternary operator here. Note that if you want to check whether a job is done there's also the job state to consider (see comments in `Constants.pm`).

Reply:
It is kind of protective programming for 'something goes wrong' scenario. E.g. if the function returns 'Not found' for some reasons - it shall mean that that something is outside of Plugin code.